packages/block-editor/src/components/off-canvas-editor/index.js

```diff
@@ -208,9 +211,48 @@ function __ExperimentalOffCanvasEditor(
 						shouldShowInnerBlocks={ shouldShowInnerBlocks }
 						selectBlockInCanvas={ selectBlockInCanvas }
 					/>
+					<tr>
+						<td>
+							<OffCanvasEditorAppender
+								rootClientId={ clientId }
+							/>
+						</td>
+					</tr>
 				</ListViewContext.Provider>
 			</TreeGrid>
 		</AsyncModeProvider>
 	);
 }
+
+function OffCanvasEditorAppender( { rootClientId } ) {
+	const { hideInserter } = useSelect(
+		( select ) => {
+			const { getTemplateLock, __unstableGetEditorMode } =
+				select( blockEditorStore );
+
+			return {
+				hideInserter:
+					!! getTemplateLock( rootClientId ) ||
+					__unstableGetEditorMode() === 'zoom-out',
```

Comment:
Do you think this logic should be handled by the inserter component itself?

Reply:
It's copied from https://github.com/WordPress/gutenberg/blob/37fc2a1dbc1671dabe3a4771b5b030814b452aae/packages/block-editor/src/components/block-list-appender/index.js#L25-L51. As we are using Inserter directly we utilise it.

Reply:
> How can I get to this case? When I am in zoomed out mode the navigation block isn't selectable...

When _either_:

1. The block is locked.
2. The editor is in zoomed out mode.

In those cases don't shown an inserter.